Journaley/Utilities/PostMarkdownParser.cs

```diff
@@ -0,0 +1,112 @@
+            var parseState = -1;
+
+            bool firstTitleParsed = false;
+
+            var lines = formattedString.Split(
+                new string[] { "\r\n", "\n" },
+                StringSplitOptions.None);
+
+            foreach (string line in lines)
+            {
+                // We first incrementally check if the line begins with <p> tag. When it does,
+                // We turn on the paragraphBuilder to append the remaining lines.
+
+                // If the checks hit a line with the </p> closing, we disable check and proceed
+                // to parse the text, add <br /> tags and then adding it to builder
+                // where it is assembled with the rest of the file.
+                if (line.Contains("<p>"))
+                {
+                    parseState = 1;
+                }
+                else if (line.Contains("</p>"))
+                {
+                    parseState = 0;
+                }
+                else if (line.Contains("<pre>"))
+                {
+                    parseState = -1;
+                }
+
+                if (parseState == 1)
+                {
+                    if (line.Contains("</p>"))
+                    {
+                        string singleLine = line;
+                        if (!firstTitleParsed)
+                        {
+                            singleLine = Regex.Replace(singleLine, @"(<p>)(.*?)(</p>)", "<h2>$2</h2>\n<br/>");
+                            firstTitleParsed = true;
+                        }
+
+                        builder.AppendLine(ReplaceMDStrikethrough(singleLine));
+                        parseState = -1;
+                    }
+                    else
+                    {
+                        paragraphBuilder.AppendLine(ReplaceMDStrikethrough(line) + "<br />");
+                    }
+                }
+                else if (parseState == 0)
+                {
+                    paragraphBuilder.AppendLine(ReplaceMDStrikethrough(line));
+
+                    builder.AppendLine(paragraphBuilder.ToString());
+                    paragraphBuilder.Clear();
+                    parseState = -1;
+                }
+                else
+                {
+                    // For <pre> tagged text, just skip check and append to builder
+                    builder.AppendLine(line);
+                }
+            }
+
+            return builder.ToString();
+        }
+        
+        /// <summary>
+        /// Replaces "~~" strikethrough Markdown tag into del HTML tags.
+        /// </summary>
+        /// <param name="parsedString">Any string that has strikethrough tags</param>
+        /// <returns>A string with del HTML tags in place of Markdown strikethrough tags.</returns>
+        private static string ReplaceMDStrikethrough(string parsedString)
+        {
+            parsedString = Regex.Replace(parsedString, @"(~~)(.*?)(~~)", "<del>$2</del>", RegexOptions.Singleline);
```

Comment:
I think having just this one line with `Multiline` option would be enough, when we pass the entire paragraph to this method. Remove the following two lines.